tests/components/dlna_dmr/conftest.py

```diff
@@ -0,0 +1,136 @@
+
+
+@pytest.fixture
+def domain_data_mock(hass: HomeAssistant) -> Iterable[Mock]:
+    """Mock the global data used by this component.
+
+    This includes network clients and library object factories. Mocking it
+    prevents network use.
+    """
+    domain_data = create_autospec(DlnaDmrData, instance=True)
+    domain_data.upnp_factory = create_autospec(
+        UpnpFactory, spec_set=True, instance=True
+    )
+
+    upnp_device = create_autospec(UpnpDevice, instance=True)
+    upnp_device.name = MOCK_DEVICE_NAME
+    upnp_device.udn = MOCK_DEVICE_UDN
+    upnp_device.device_url = MOCK_DEVICE_LOCATION
+    upnp_device.device_type = "urn:schemas-upnp-org:device:MediaRenderer:1"
+    upnp_device.available = True
+    seal(upnp_device)
+    domain_data.upnp_factory.async_create_device.return_value = upnp_device
+
+    domain_data.unmigrated_config = {}
+
+    with patch.dict(hass.data, {DLNA_DOMAIN: domain_data}):
+        yield domain_data
+
+    # Make sure the event notifiers are released
+    assert (
+        domain_data.async_get_event_notifier.await_count
+        == domain_data.async_release_event_notifier.await_count
+    )
+
+
+@pytest.fixture
+def config_entry_mock() -> Iterable[MockConfigEntry]:
+    """Mock a config entry for this platform."""
+    mock_entry = MockConfigEntry(
+        unique_id=MOCK_DEVICE_UDN,
+        domain=DLNA_DOMAIN,
+        data={
+            CONF_URL: MOCK_DEVICE_LOCATION,
+            CONF_DEVICE_ID: MOCK_DEVICE_UDN,
+            CONF_TYPE: MOCK_DEVICE_TYPE,
+        },
+        title=MOCK_DEVICE_NAME,
+        options={},
+    )
+    yield mock_entry
+
+
+@pytest.fixture
+def dmr_device_mock(domain_data_mock: Mock) -> Iterable[Mock]:
+    """Mock the async_upnp_client DMR device, initially connected."""
+    with patch(
+        "homeassistant.components.dlna_dmr.media_player.DmrDevice", autospec=True
+    ) as constructor:
+        device = constructor.return_value
+        device.on_event = None
+        device.device = domain_data_mock.upnp_factory.async_create_device.return_value
+        device.media_image_url = "http://192.88.99.20:8200/AlbumArt/2624-17620.jpg"
+        device.udn = "device_udn"
+        device.manufacturer = "device_manufacturer"
+        device.model_name = "device_model_name"
+        device.name = "device_name"
+
+        yield device
+
+        # Make sure the device is disconnected
+        assert (
+            device.async_subscribe_services.await_count
+            == device.async_unsubscribe_services.await_count
+        )
+
+        assert device.on_event is None
+
+
+@pytest.fixture(name="skip_notifications", autouse=True)
+def skip_notifications_fixture() -> Iterable[None]:
+    """Skip notification calls."""
+    with patch("homeassistant.components.persistent_notification.async_create"), patch(
+        "homeassistant.components.persistent_notification.async_dismiss"
+    ):
+        yield
+
+
+@pytest.fixture(autouse=True)
```

Comment:
This looks similar to https://github.com/home-assistant/core/pull/56431

We should clean this up in a new PR once both merge.